### PR TITLE
test(node-sdk): Add basic unit tests and coverage

### PR DESCRIPTION
**Pull request by bhelx**

### CODE REVIEW

File: node/src/index.ts [hunk 1]
```diff
@@ -76,12 +74,12 @@ export function extismVersion(): string {
 
 // @ts-ignore
 const pluginRegistry = new FinalizationRegistry(({ id, pointer }) => {
-    lib.extism_plugin_free(pointer, id);
+    if (id && pointer) lib.extism_plugin_free(pointer, id);
```
Comment (bhelx) [inline node/src/index.ts:77]: Found a bug: Guard for the case when the pointers are already null

File: node/src/index.ts [hunk 2]
```diff
@@ -106,14 +104,13 @@ export type ManifestWasm = ManifestWasmFile | ManifestWasmData;
 
 export type Manifest = {
     wasm: Array<ManifestWasm>;
-    memory: ManifestMemory;
+    memory?: ManifestMemory;
```
Comment (bhelx) [inline node/src/index.ts:107]: mis-marked this. should have been optional

File: node/package.json [hunk 3]
```diff
@@ -14,16 +14,16 @@
   "private": false,
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
-  "type": "module",
```
Comment (bhelx) [inline node/package.json:17]: This was causing some of the TS tooling to expect an `mjs` or `cjs` file. I tested using this as a dep and didn't seem to need this.

**Comment by bhelx** [comment]
Interesting failure in the CI:

```
#
# Fatal error in , line 0
# Check failed: result.second.
#
#
#
#FailureMessage Object: 0x7ffc739a6b50
 1: 0xb6cab1  [node]
 2: 0x1bef8b4 V8_Fatal(char const*, ...) [node]
 3: 0xfb2ef1 v8::internal::GlobalBackingStoreRegistry::Register(std::shared_ptr<v8::internal::BackingStore>) [node]
 4: 0xd04968 v8::ArrayBuffer::GetBackingStore() [node]
 5: 0xab8d00 napi_get_typedarray_info [node]
 6: 0x7fc2440103ac  [/home/runner/work/extism/extism/node/node_modules/ref-napi/prebuilds/linux-x64/node.napi.node]
 7: 0x7fc244010b48  [/home/runner/work/extism/extism/node/node_modules/ref-napi/prebuilds/linux-x64/node.napi.node]
 8: 0x7fc244012831  [/home/runner/work/extism/extism/node/node_modules/ref-napi/prebuilds/linux-x64/node.napi.node]
 9: 0x7fc2440184ab Napi::details::CallbackData<void (*)(Napi::CallbackInfo const&), void>::Wrapper(napi_env__*, napi_callback_info__*) [/home/runner/work/extism/extism/node/node_modules/ref-napi/prebuilds/linux-x64/node.napi.node]
10: 0xaaf64d  [node]
11: 0xd3976e  [node]
12: 0xd3ab8f v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [node]
13: 0x15d55d9  [node]
Trace/breakpoint trap (core dumped)
Error: Process completed with exit code 133.
```

Will check tomorrow

**Review by zshipko — APPROVED** [review]

**Comment by zshipko** [comment]
I ran into that failure again testing other languages on CI, seems like it might be worth looking into. I will see if I can reproduce it.

**Comment by bhelx** [comment]
Feel free to revert if it is blocking you

**Comment by zshipko** [comment]
Cool I'll wait and see if it keeps popping up before reverting anything.

**Pull request merged**